MapboxNavigation/TopBannerViewController.swift

```diff
@@ -0,0 +1,433 @@
+        view.bottomAnchor.constraint(equalTo: stepsContainer.bottomAnchor)
+        ]
+        return constraints
+    }()
+    
+    lazy var stepsContainerHide: [NSLayoutConstraint] = {
+        let constraints = [
+        stepsContainer.bottomAnchor.constraint(equalTo: informationStackView.topAnchor),
+        informationStackBottomPin
+        ]
+        return constraints
+    }()
+    
+    lazy var informationStackBottomPin: NSLayoutConstraint = view.bottomAnchor.constraint(equalTo: informationStackView.bottomAnchor)
+
+    lazy var informationStackView = UIStackView(orientation: .vertical, autoLayout: true)
+    
+    lazy var instructionsBannerView: InstructionsBannerView = {
+        let banner: InstructionsBannerView = .forAutoLayout()
+        banner.heightAnchor.constraint(equalToConstant: instructionsBannerHeight).isActive = true
+        banner.delegate = self
+        banner.swipeable = true
+        return banner
+    }()
+    
+    lazy var lanesView: LanesView = .forAutoLayout(hidden: true)
+    lazy var nextBannerView: NextBannerView = .forAutoLayout(hidden: true)
+    lazy var statusView: StatusView = {
+        let view: StatusView = .forAutoLayout()
+        view.delegate = delegate
+        view.isHidden = true
+        return view
+    }()
+    
+    private let instructionsBannerHeight: CGFloat = 100.0
+    
+    private var informationChildren: [UIView] {
+        return [instructionsBannerView] + secondaryChildren
+    }
+    private var secondaryChildren: [UIView] {
+        return  [lanesView, nextBannerView, statusView]
+    }
+    
+    public var isDisplayingPreviewInstructions: Bool {
+        return previewInstructionsView != nil
+    }
+    
+    private(set) public var isDisplayingSteps: Bool = false
+    
+    
+    private(set) var previewSteps: [RouteStep]?
+    private(set) var currentPreviewStep: (RouteStep, Int)?
+    
+    private(set) var previewInstructionsView: StepInstructionsView?
+    
+    required public init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        commonInit()
+    }
+    
+    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
+        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
+        commonInit()
+    }
+    
+    convenience init(delegate: TopBannerViewControllerDelegate) {
+        self.init(nibName: nil, bundle: nil)
+        self.delegate = delegate
+        statusView.delegate = delegate
+    }
+    
+    func commonInit() {
+ 
+    }
+    
+    
+    override open func viewDidLoad() {
+        view.backgroundColor = .clear
+        super.viewDidLoad()
+        setupViews()
+        addConstraints()
+        setupInformationStackView()
+    }
+    
+    
+    private func setupViews() {
+        topPaddingView.accessibilityIdentifier = "topPaddingView"
+        let children = [stepsContainer, topPaddingView, informationStackView]
+        children.forEach(view.addSubview(_:))
```

Comment:
Nit: Seems like it would be clearer just to say:

```swift
view.addSubview(stepsContainer)
view.addSubview(topPaddingView)
view.addSubview(informationStackView)
```